backend/api/users.controller.js

```diff
@@ -57,4 +58,34 @@ module.exports = class UserController {
       res.status(500).json({ error: e.message });
     }
   }
+
+  static async apiUpdateUser(req, res, next) {
+    //update what parts? for testing any..
+    try {
+      const id = req.body._id;
+      const user_id = req.body.user_id;
+      const username = req.body.username;
+      const name = req.body.name;
+      const email = req.body.email;
+      console.log(`mongodb id: ${id}, and userid: ${user_id}`);
+      console.log(username, name, email);
+      const userResponse = await UsersDAO.updateUser(
+        user_id,
+        id,
+        username,
+        name,
+        email
+      );
+      var { error } = userResponse;
+      if (error) {
+        res.status(400).json({ error });

```

Comment:
If we pass empty fields, what is the error message?
Be careful to not dump DB errors in production.
If this dumps the DB errors in prod, we could use a library to validate input.
We don't need to implement this now though.

Reply:
I will have to reinvestigate, I put this code to the side because it no longer lined up with the MVP, and cannot remember exactly....